public/js/components/Frames.js

```diff
@@ -8,22 +9,24 @@ const actions = require("../actions");
 const { endTruncateStr } = require("../utils/utils");
 const { getFilename } = require("../utils/source");
 const { getFrames, getSelectedFrame, getSource } = require("../selectors");
+const { List } = require("immutable");
+const classNames = require("classnames");
 
 if (typeof window == "object") {
   require("./Frames.css");
 }
 
 const NUM_FRAMES_SHOWN = 7;
 
-function renderFrameTitle(frame) {
-  return div({ className: "title" }, endTruncateStr(frame.displayName, 40));
+function renderFrameTitle({ displayName }: Object) {
```

Comment:
This doesn't work. https://github.com/facebook/flow/issues/183 has been tracking broken support for destructuring in Flow for a while.

The problem is `{ source: Source}` transpiles to `var Source = object.source`.

Instead, I tried:
```javascript
function renderFrameTitle({ displayName }: { displayName: string }) {
```

Unfortunately, this doesn't typecheck:
```
public/js/components/Frames.js:67
 67:       renderFrameTitle(frame),
           ^^^^^^^^^^^^^^^^^^^^^^^ function call
 25: function renderFrameTitle({ displayName }: { displayName: string }) {
                                                ^^^^^^^^^^^^^^^^^^^^^^^ property `displayName`. Property not found in
 67:       renderFrameTitle(frame),
                            ^^^^^ object type
```

I'll see what I can figure out by creating a `frame` type.